src/components/Drawer/components/Drawer.tsx

```diff
@@ -0,0 +1,216 @@
+        onTransitionIn,
+        onTransitionInComplete,
+        onTransitionOut,
+        onTransitionOutComplete,
+        context,
+    });
+
+    const dismiss = useDismiss(context, {
+        enabled: !disableEscapeKeyDown || !disableOutsideClick,
+        outsidePress: (event) => {
+            if (disableOutsideClick) {
+                return false;
+            }
+
+            const isOwnOutsideClick =
+                (event.target as HTMLElement).closest(`.${b()}`) === overlayRef.current;
+
+            if (!isOwnOutsideClick) {
+                return false;
+            }
+
+            return true;
+        },
+        escapeKey: !disableEscapeKeyDown,
+    });
+
+    const role = useRole(context, {role: 'dialog'});
+    const {getFloatingProps} = useInteractions([dismiss, role]);
+    const handleFloatingRef = useForkRef<HTMLDivElement>(refs.setFloating, floatingRef);
+
+    const portal =
+        isMounted || keepMounted ? (
+            <Portal container={container} disablePortal={disablePortal}>
+                <FloatingOverlay
+                    style={style}
+                    ref={overlayRef}
+                    aria-modal="true"
+                    className={b(
+                        {
+                            open: isOpen,
+                            'hide-veil': hideVeil,
+                            'skip-animation': !showInitialAnimation && isInitialRender,
+                        },
+                        className,
+                    )}
+                    data-qa={qa}
+                    data-floating-ui-status={status}
+                    data-transiting={isTransitionInProgress}
+                    lockScroll={!disableBodyScrollLock}
+                >
+                    <div className={b('veil', {hidden: hideVeil})} />
+                    <FloatingFocusManager
+                        context={context}
+                        disabled={!isMounted}
+                        modal={isMounted}
+                        initialFocus={initialFocus ?? refs.floating}
+                        returnFocus={returnFocus}
+                        visuallyHiddenDismiss={disableVisuallyHiddenDismiss ? false : i18n('close')}
+                        restoreFocus={true}
+                    >
+                        <DrawerItem
+                            ref={handleFloatingRef}
+                            open={isOpen}
+                            direction={direction}
+                            className={contentClassName}
+                            resizable={resizable}
+                            size={size}
+                            onResize={onResize}
+                            onResizeStart={onResizeStart}
+                            onResizeEnd={onResizeEnd}
+                            minResizeWidth={minSize}
+                            maxResizeWidth={maxSize}
+                            {...getFloatingProps()}
+                            {...restProps}
```

Comment:
We should wrap it into `filterDOMProps`. See Modal [implementaion](https://github.com/gravity-ui/uikit/blob/main/src/components/Modal/Modal.tsx#L282)